File: src/main/java/kr/dogfoot/hwpxlib/object/content/section_xml/SectionXMLFile.java
package kr.dogfoot.hwpxlib.object.content.section_xml;

import kr.dogfoot.hwpxlib.object.common.ObjectType;

/**
 * Section.xml 파일
 */
public class SectionXMLFile extends ParaListCore {
    public SectionXMLFile() {
    }

    @Override
    public ObjectType _objectType() {
        return ObjectType.hs_sec;
    }

    public SectionXMLFile clone() {
        SectionXMLFile cloned = new SectionXMLFile();
        cloned.copyFrom(this);
        return cloned;
    }

    public void copyFrom(SectionXMLFile from) {
        super.copyFrom(from);
    }
}

File: js/src/object/content/section_xml/SectionXMLFile.js
import { ParaListCore } from './ParaListCore.js';
import { ObjectType } from '../../common/ObjectType.js';

/**
 * Section XML file representation (section0.xml, section1.xml, etc.)
 */
export class SectionXMLFile extends ParaListCore {
  constructor() {
    super();
  }

  _objectType() {
    return ObjectType.hs_sec;
  }

  clone() {
    const cloned = new SectionXMLFile();
    cloned.copyFrom(this);
    return cloned;
  }

  copyFrom(from) {
    this._copyFromParaListCore(from);
  }
}

export default SectionXMLFile;
